badge_server/main.py

```diff
@@ -76,518 +58,196 @@ def _get_pair_status_for_packages(pkg_sets):
-
-                for res in results:
-                    res_item = res[0]
-                    status = res_item.get('result')
-                    package = res_item.get('packages')[1]
-                    if status != 'SUCCESS':
-                        # Ignore the package that not support for given py_ver
-                        if package in \
-                                configs.PKG_PY_VERSION_NOT_SUPPORTED.get(
-                                py_ver):
-                            continue
-
-                        # Ignore the package that are not self compatible
-                        self_status = _get_result_from_cache(
-                            package_name=package_name,
-                            badge_type=badge_utils.BadgeType.SELF_COMP_BADGE)
-                        if self_status[py_version]['status'] not in [
-                                'SUCCESS', 'CALCULATING']:
-                            continue
-                        # Status showing one of the check failures
-                        version_and_res[
-                            py_version]['status'] = res_item.get('result')
-                        description = res_item.get('description')
-                        details = badge_utils.EMPTY_DETAILS if description \
-                            is None else description
-                        version_and_res[
-                            py_version]['details'][package] = details
-            version_and_res['timestamp'] = datetime.datetime.now().strftime(
-                badge_utils.TIMESTAMP_FORMAT)
-            result = version_and_res
-
-        # Write the result to Cloud Datastore
-        cache.set('{}_google_comp_badge'.format(package_key), result)
-
-    google_comp_res = cache.get('{}_google_comp_badge'.format(package_key))
-
-    if not utils._is_package_in_whitelist([package_name]):
-        google_comp_res = badge_utils._build_default_result(
-            badge_type=badge_utils.BadgeType.GOOGLE_COMP_BADGE,
-            status='UNKNOWN',
-            details={})
-
-    if google_comp_res is None:
-        details = badge_utils._build_default_result(
-            badge_type=badge_utils.BadgeType.GOOGLE_COMP_BADGE,
-            status='CALCULATING',
-            details={})
-    else:
-        details = google_comp_res
-
-    # Run the check if google_comp_res is None or forced to populate the cache
-    # or the cache is outdated.
-    if google_comp_res is None or force_run_check is not None:
-        threading.Thread(target=run_check).start()
-    elif google_comp_res is not None:
-        timestamp = google_comp_res.get('timestamp')
-        if not badge_utils._is_github_cache_valid(timestamp):
-            threading.Thread(target=run_check).start()
-
-    badge = badge_utils._get_badge(details, badge_name)
     response = flask.make_response(badge)
     response.content_type = badge_utils.SVG_CONTENT_TYPE
     response.headers['Cache-Control'] = 'no-cache'
     response.add_etag()
```

Comment:
I didn't write this code; maybe @liyanhui1228 can provide context?

Reply:
I have a vague memory that without this line the badge image refresh won't work and github will still cache the image. Did you try that?

Reply:
The etag is required to preventing the caching or the Cache-Control header is required.

I took a quick look at RFC-2616 (https://tools.ietf.org/html/rfc2616#section-13.4). So, for the Cache-Control header, a reasonable comment would be:

```python
# https://tools.ietf.org/html/rfc2616#section-13.4 allows success responses to be cached if
# no `Cache-Control` header is set. Since the content of the image is frequently updated,
# caching is explicitly disabled to force the client/cache to refetch the content on every
# request.
```

You can update the comment if you end up changing the value of the header.

Reply:
done